go/vt/mysqlctl/azblobbackupstorage/azblob.go

```diff
@@ -166,7 +185,10 @@ func (bh *AZBlobBackupHandle) ReadFile(ctx context.Context, filename string) (io
 	return resp.Body(azblob.RetryReaderOptions{
 		MaxRetryRequests: defaultRetryCount,
 		NotifyFailedRead: func(failureCount int, lastError error, offset int64, count int64, willRetry bool) {
-			bh.errors.RecordError(lastError)
+			log.Infof("ReadFile: [azblob] container: %s, directory: %s, filename: %s, error: %v", *containerName, objName(bh.dir, ""), filename, lastError)
+			if !willRetry {
+				bh.errors.RecordError(lastError)
```

Comment:
We already have a code path that's supposed to catch non-retryable errors: we should get an error while we're actually trying to read from the stream. It seems risky and unnecessary to add in a second path that tries to catch and report the same error.